pkg/reconciler/event.go

```diff
@@ -0,0 +1,76 @@
+package reconciler
+
+import (
+	"fmt"
+	"path/filepath"
+	"strconv"
+
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/apis/pipelinesascode"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/params/info"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/provider"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/provider/bitbucketcloud"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/provider/bitbucketserver"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/provider/github"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/provider/gitlab"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+)
+
+func detectProvider(pr *v1beta1.PipelineRun) (provider.Interface, error) {
+	gitProvider, ok := pr.GetLabels()[filepath.Join(pipelinesascode.GroupName, "git-provider")]
+	if !ok {
+		return nil, fmt.Errorf("failed to detect git provider for pipleinerun %s : git-provider label not found", pr.GetName())
+	}
+
+	var provider provider.Interface
```

Comment:
can it be common with adapter?

Reply:
didn't get it 🤔
what do you mean ? sharing the obj?

Reply:
yeah never mind! 😅